tests/test_api.py

```diff
@@ -446,3 +447,52 @@ def test_pop():
         respx.get("https://foo.bar/", alias="foobar")
         request_pattern = respx.pop("foobar")
         assert request_pattern.url == "https://foo.bar/"
+
+
+@respx.mock
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "url,params,call_url,call_params",
+    [
+        ("https://foo/", "foo=bar", "https://foo/", "foo=bar"),
+        ("https://foo/", b"foo=bar", "https://foo/", b"foo=bar"),
+        ("https://foo/", [("foo", "bar")], "https://foo/", [("foo", "bar")]),
+        ("https://foo/", {"foo": "bar"}, "https://foo/", {"foo": "bar"}),
+        ("https://foo?foo=bar", "baz=qux", "https://foo?foo=bar", "baz=qux"),
+        ("https://foo?foo=bar", "baz=qux", "https://foo?foo=bar&baz=qux", None),
+        (re.compile(r"https://foo/(\w+)/"), "foo=bar", "https://foo/bar/", "foo=bar"),
+        (httpx.URL("https://foo/"), "foo=bar", "https://foo/", "foo=bar"),
+        (
+            httpx.URL("https://foo?foo=bar"),
+            "baz=qux",
+            "https://foo?foo=bar&baz=qux",
+            None,
+        ),
+    ],
+)
+async def test_params_match(client, url, params, call_url, call_params):
+    respx.get(url, params=params, content="spam spam")
+    response = await client.get(call_url, params=call_params)
+    assert response.text == "spam spam"
+
+
+# @pytest.mark.asyncio
```

Comment:
These tests breaks `test_mock.test_decorator`. Seems like the context manager doesn't release internal state in respx, which may confuse other tests.

Reply:
Weird, sounds like a bug maybe?

Is it `base_url` that is set in this test and then kept for other tests resulting in fail?

What's different from the working `test_mock.test_base_url`?